### PR TITLE
Remove unused/usafe Get/Set on Derived data

### PR DESCRIPTION
**Pull request by janezpodhostnik**
ref: https://github.com/onflow/cadence/issues/1684

### CODE REVIEW

**Review by pattyshack — APPROVED** [review]

File: fvm/environment/programs.go [hunk 1]
```diff
@@ -428,6 +171,96 @@ func (programs *Programs) cacheMiss() {
 	programs.metrics.RuntimeTransactionProgramsCacheMiss()
 }
 
+// programLoader is used to load a program from a location.
+type programLoader struct {
+	loadFunc        func() (*interpreter.Program, error)
+	dependencyStack *dependencyStack
+	called          bool
```
Comment (zhangchiqing) [inline fvm/environment/programs.go:178]: is the programLoader used only for loading a single address?

Is it possible the same loader being used by loading multiple addresses?  In that case, the `called` just means it's called for loading the first address. 

Do we call `loadFunc` twice if loading the same address twice?

Comment (janezpodhostnik) [reply fvm/environment/programs.go:178]: loadFunc is being called only once. and is not reused in any way. Let me see if I can make this clearer in the code.

Comment (janezpodhostnik) [reply fvm/environment/programs.go:178]: Added a panic and a comment if called multiple times.

Comment (janezpodhostnik) [reply fvm/environment/programs.go]: (this comment is actually for https://github.com/onflow/flow-go/pull/4002 i just forgot to change the base)

**Comment by janezpodhostnik** [comment]
bors try

**Review by ramtinms — APPROVED** [review]

**Comment by janezpodhostnik** [comment]
bors merge

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**